mpas_analysis/ocean/meridional_overturning_circulation.py

```diff
@@ -0,0 +1,445 @@
+
+    mocAnalysisMemberFlag = namelist.get(
+        'config_am_mocstreamfunction_enable')
+
+    calendar = namelist.get('config_calendar_type')
+    simulationStartTime = get_simulation_start_time(streams)
+
+    # get a list of timeSeriesStats output files from the streams file,
+    # reading only those that are between the start and end dates
+    streamName = streams.find_stream(streamMap['timeSeriesStats'])
+    #   First a list necessary for the streamfunctionMOC climatology
+    startDateClimo = config.get('climatology', 'startDate')
+    endDateClimo = config.get('climatology', 'endDate')
+    inputFilesClimo = streams.readpath(streamName,
+                                       startDate=startDateClimo,
+                                       endDate=endDateClimo,
+                                       calendar=calendar)
+    print '\n  List of files for climatologies:\n{} through {}'.format(
+           inputFilesClimo[0], inputFilesClimo[-1])
+    startYearClimo = config.getint('climatology', 'startYear')
+    endYearClimo = config.getint('climatology', 'endYear')
+
+    plotsDirectory = buildConfigFullPath(config, 'output', 'plotsSubdirectory')
+    mainRunName = config.get('runs', 'mainRunName')
+    #   Then a list necessary for the streamfunctionMOC Atlantic timeseries
+    startDateTseries = config.get('timeSeries', 'startDate')
+    endDateTseries = config.get('timeSeries', 'endDate')
+    inputFilesTseries = streams.readpath(streamName,
+                                         startDate=startDateTseries,
+                                         endDate=endDateTseries,
+                                         calendar=calendar)
+    print '  List of files for timeSeries:\n{} through {}'.format(
+           inputFilesTseries[0], inputFilesTseries[-1])
+    startYearTseries = config.getint('timeSeries', 'startYear')
+    endYearTseries = config.getint('timeSeries', 'endYear')
+
+    try:
+        restartFile = streams.readpath('restart')[0]
+    except ValueError:
+        raise IOError('No MPAS-O restart file found: need at least one '
+                      'restart file for MOC calculation')
+
+    sectionNameClimo = 'streamfunctionMOC'
+    sectionNameTseries = 'timeSeriesMOC'
+
+    # Plotting variables
+    colormapName = plt.get_cmap(config.get(sectionNameClimo, 'colormapName'))
+    colormapIndices = config.getExpression(sectionNameClimo, 'colormapIndices')
+    colormapName = cols.ListedColormap(colormapName(colormapIndices),
+                                       'colormapName')
+    colorbarLabel = '[Sv]'
+    movingAveragePoints = config.getint(sectionNameTseries,
+                                        'movingAveragePoints')
+
+    # Check whether MOC Analysis Member is enabled
+    if mocAnalysisMemberFlag == '.true.':
+
+        # Add a moc_analisysMember_processing
+        print '*** MOC Analysis Member is on ***'
+
+    else:
+
+        # Load mesh related variables
```

Comment:
> what do I do with the variables that I need in both the main function and the helper functions?

If they're relatively trivial to extract from config, I would redefine them rather than passing them as arguments to your helper functions.  If they're complicated to compute (like data sets -- basically anything that takes more than one or 2 lines to get back), it would be better to pass them in.